### ISSUE TITLE
multipart_upload_test fails

### ISSUE THREAD

**Issue by Waitak** [zino-app/graphql-flutter#525]
**Describe the bug**
`graphql/test/multipart_upload_test` fails at line 161 because `application/json` is returned instead of `multipart/form-data`.

**To Reproduce**
1. cd graphql-flutter/packages/graphql
1. pub run test_coverage
1. Interrupt as soon as `.test_coverage.dart` is generated.
1. Run test manually using:
    dart --pause-isolates-on-exit \\
        --enable_asserts \\
        --enable-vm-service=8787 test/.test_coverage.dart

The relevant section of the test output is as follows:

```
00:01 +64 -1: upload upload success [E]
  Expected: 'multipart/form-data'
    Actual: 'application/json'
     Which: is different.
            Expected: multipart/ ...
              Actual: applicatio ...
                      ^
             Differ at offset 0

  package:test_api/src/frontend/expect.dart 153:30              fail
  package:test_api/src/frontend/expect.dart 147:3               _expect
  package:test_api/src/frontend/expect.dart 59:3                expect
  test/multipart_upload_test.dart 161:7                         main.<fn>.<fn>
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Version 2.1.1-beta.5

**Additional context**
- [Test coverage full output](https://github.com/zino-app/graphql-flutter/files/4027066/test_coverage.txt)

**Comment by knaeckeKami** [comment]
Likely related:
https://github.com/dart-lang/http/issues/351
Might be a bug in http 0.12.0+3 and fixed in 0.12.0+4?

**Comment by mainawycliffe** [comment]
Amazing work @knaeckeKami.

**Comment by kulture-rob** [comment]
I had a feeling it had to do with the HTTP module, but their changelog just had documentation changes the past couple of releases. Thanks for finding that!!

**Comment by mainawycliffe** [comment]
This issue has been fixed by #530, should be released to beta over the next 24 hours.
